vllm/worker/model_runner.py

```diff
@@ -871,6 +872,11 @@ def capture_model(self, kv_caches: List[torch.Tensor]) -> None:
         seq_lens = torch.ones(max_batch_size, dtype=torch.int32).cuda()
         block_tables = torch.from_numpy(self.graph_block_tables).cuda()
 
+        # Prepare buffer for outputs. These will be reused for all batch sizes.
+        hidden_states = torch.zeros(
+            (max_batch_size, self.model_config.get_output_size()),
+            dtype=self.model_config.dtype).cuda()
```

Comment:
Why don't we reuse the output buffer of the first CUDA Graph (which is for the largest batch size)? Arguably, this will be more reliable than the current implementation.

Reply:
That's doable. Although the code might look more ugly.

Reply:
While it's debatable, I personally think it's more reliable than the current implementations which relies on HF config and `gc`. If the user has a custom model, the current implementation might break.

Reply:
fixed in [dbd5248](https://github.com/vllm-project/vllm/pull/5074/commits/dbd5248db37e276c409f1c65eacde21398b2f8c9)